### ISSUE TITLE
Adding one constraint per variable raises Argument Exception

### ISSUE THREAD

**Issue by balalofernandez** [BrannonKing/NLoptNet#30]
I have _n_ variables that I need to make binary, therefore I'm writing a simple constraint `x[i]-x[i]^2=0`. I code it like this:
```
for (int i = 0; i < n; i++)
{
    solver.AddEqualZeroConstraint(variables =>
    {
        return Math.Pow(variables[i], 2) - variables[i];
    },0.01);
}
````
However, when I run it I get this exception:

> System.ArgumentException: 'An item with the same key has already been added. Key: System.Func`2[System.Double[],System.Double]'

I'm adding the variable twice to the `_funcCache` dictionary, anyone knows how to solve it?

**Comment by BrannonKing** [comment]
That's definitely a bug in NLoptNet. It should be using a list of tuples instead of a dictionary. The _funcCache is just to ensure that the internal anonymous functions don't get garbage-collected before they're used. Also, don't use Math.Pow for squaring a variable.

**Comment by BrannonKing** [comment]
Fixed in 1.4.2
